src/sqlparser/src/tokenizer.rs

```diff
@@ -589,6 +591,13 @@ impl<'a> Tokenizer<'a> {
                         _ => Ok(Some(Token::Colon)),
                     }
                 }
+                '$' => {
+                    if let Some(parameter) = self.tokenize_parameter(chars) {
+                        Ok(Some(parameter))
+                    } else {
+                        Ok(Some(Token::Char('$')))
+                    }
+                }
```

Comment:
FYI in PostgreSQL `$` can also start a [dollar-quoted string constant](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING) (upstream https://github.com/sqlparser-rs/sqlparser-rs/pull/772).

But extending the support later would not break common use cases.